src/Alpha_complex/include/gudhi/Alpha_complex.h

```diff
@@ -376,10 +376,10 @@ class Alpha_complex {
    *
    * Initialization can be launched once.
    *
-   * \note Weighted Alpha complex can have negative filtration values. If `output_squared_values` is set to `false`,
-   * filtration values will be `NaN` in this case.
+   * \exception std::invalid_argument In case of a Weighted Alpha complex with `Output_squared_values` set to `false`,
+   * as Weighted Alpha complex can have negative filtration values.
```

Comment:
Radical, but I am fine with the new proposition (even though, the value should be the other way around, as the filtration value is not squared (couldn't be negative otherwise), but technically it would not fit with the default value of the non weighted case). But here I would write "..., as the filtration values of a Weighted Alpha complex results from a power product and not from an euclidean distance. Therefore this template argument should be kept to the default value." instead of "..., as Weighted Alpha complex can have negative filtration values."